src/opening-period/page/CreateNewOpeningPeriodPage.tsx

```diff
@@ -198,99 +67,40 @@ export default function CreateNewOpeningPeriodPage({
-          </h2>
+          <ResourceInfoSubTitle text="Toimipisteen aukiolojakson lisäys" />
         </ResourceTitle>
         <ResourceAddress resource={resource} />
       </ResourceInfo>
-      <form
-        id="add-new-opening-period-form"
-        data-test="add-new-opening-period-form"
-        className="add-new-opening-period-form"
-        onSubmit={handleSubmit(onSubmit)}>
-        <section className="form-section">
-          <OpeningPeriodDescription register={register} errors={errors} />
-        </section>
-        <section className="form-section">
-          <h3 className="opening-period-section-title">Ajanjakso</h3>
-          <section className="opening-period-time-period">
-            <Datepicker
-              id="openingPeriodBeginDate"
-              dataTest="openingPeriodBeginDate"
-              labelText="Alkaa"
-              onChange={(value): void => setPeriodBeginDate(value || null)}
-              value={periodBeginDate}
-              registerFn={register}
-              required
-            />
-            <p className="dash-between-begin-and-end-date">—</p>
-            <Datepicker
-              id="openingPeriodEndDate"
-              dataTest="openingPeriodEndDate"
-              labelText="Päättyy"
-              onChange={(value): void => setPeriodEndDate(value || null)}
-              value={periodEndDate}
-              registerFn={register}
-              required
-            />
-          </section>
-        </section>
-        <section className="form-section time-span-group">
-          <div className="time-span-list-container">
-            <h3 className="opening-period-section-title">Aukioloajat</h3>
-            {fields.map((item, index) => (
-              <TimeSpan
-                resourceStateOptions={resourceStateOptions}
-                setValue={setValue}
-                register={register}
-                key={`time-span-${index}`}
-                index={index}
-                remove={remove}
-              />
-            ))}
-          </div>
-          <SecondaryButton
-            dataTest="add-new-time-span-button"
-            onClick={(): void => append({})}
-            className="add-new-opening-period-final-action-button add-new-time-span-button">
-            + Lisää aukioloaika
-          </SecondaryButton>
-        </section>
-        <div className="add-new-opening-period-final-action-row-container">
-          <PrimaryButton
-            dataTest="publish-new-opening-period-button"
-            className="add-new-opening-period-final-action-button"
-            type="submit">
-            Julkaise
-          </PrimaryButton>
-          <SecondaryButton
-            className="add-new-opening-period-final-action-button"
-            onClick={(): void => history.push(`/resource/${resourceId}`)}>
-            Peruuta ja palaa
-          </SecondaryButton>
-        </div>
-      </form>
+      {resource && datePeriodOptions && (
```

Comment:
I think these should already exist if the app gets this far. But is it because of typescript that we have to do this check?

Reply:
Partly yes, partly because of state-hooks 😁